libmscore/read301.cpp

```diff
@@ -336,5 +337,183 @@ Score::FileError MasterScore::read301(XmlReader& e)
+      {
+      QPageSize::PageSizeId psid = QPageSize::id(size,
+                                                 QPageSize::Inch,
+                                                 QPageSize::FuzzyOrientationMatch);
+      int id = int(psid);
+      if (psid != QPageSize::Custom &&
+          MScore::sizesMetric.find(id)   == MScore::sizesMetric.end() &&
+          MScore::sizesImperial.find(id) == MScore::sizesImperial.end() &&
+          MScore::sizesOther.find(id)    == MScore::sizesOther.end())
+            {
+            psid = QPageSize::Custom;
+            }
+      return psid;
+      }
+
+//------------------------------------------------------------------------------
+//   MStyle::toPageLayout301
+//------------------------------------------------------------------------------
 
+void MStyle::toPageLayout301()
+      {
+      double factor, w, h, p, left, right, top, bottom, eLeft;
+      int idxUnit;
+      bool isTwo, isLandscape;
+      QMarginsF oddMarg, evenMarg;
+
+      w = fuzzyDefault(Sid::pageWidth);
+      h = fuzzyDefault(Sid::pageHeight);
+      p = fuzzyDefault(Sid::pagePrintableWidth);
+
+      idxUnit = MScore::unitsValue();
+      factor  = pageUnits[idxUnit].inchFactor(); // convert inches to user units
+      top     = fuzzyDefault(Sid::pageOddTopMargin) / factor;
+      bottom  = fuzzyDefault(Sid::pageOddBottomMargin) / factor;
+      left    = fuzzyDefault(Sid::pageOddLeftMargin) / factor;
+      eLeft   = fuzzyDefault(Sid::pageEvenLeftMargin) / factor;
+      isTwo   = value(Sid::pageTwosided).toBool();
+      right   = isTwo ? eLeft : ((w - p) / factor) - eLeft;
+      oddMarg = QMarginsF(left, top, right, bottom);
+      top     = fuzzyDefault(Sid::pageEvenTopMargin) / factor;
+      bottom  = fuzzyDefault(Sid::pageEvenBottomMargin) / factor;
+      if (isTwo)
+            right = left; // two-sided = mirror image margins
+      evenMarg = QMarginsF(eLeft, top, right, bottom);
+
+      isLandscape = (w > h);
+      QSizeF sz = QSizeF(w, h);
+      QPageSize::PageSizeId psid = pageSizeId301(sz);
+      if (psid != QPageSize::Custom) {
+            QSizeF size = _pageSize.definitionSize();
+            isLandscape = (isLandscape != (size.width() > size.height()));
+            _pageSize   = QPageSize(psid);
+            }
+      else
+            _pageSize = QPageSize(QSizeF(w / factor, h / factor),
+                                  QPageSize::Unit(idxUnit),
+                                  QPageSize::name(psid),
+                                  QPageSize::ExactMatch);
+      _pageOdd = MPageLayout();
+      _pageOdd.setPageSize(_pageSize);
+      _pageOdd.setUnits(QPageLayout::Unit(idxUnit));
+      _pageOdd.setOrientation(isLandscape ? QPageLayout::Landscape
+                                          : QPageLayout::Portrait);
+      _pageOdd.setMargins(oddMarg);
+      if (MScore::testMode && _pageOdd.margins().left() == 0) { /// Travis workaround
```

Comment:
The basic problem in Travis is that the QMarginsF values are all zero.  It is only a problem on Travis, though no one has tested it besides me on my Ubuntu partition.  But the code makes sense, the Travis log file entries do not.

Reply:
I just tried this for compat114, measure, and split mtests.  It acts normally, and does not reproduce the Travis behavior.  My qDebug statements are never executed.

Reply:
I'm very concerned and curious about this, so I'll fiddle around and verify a bit further, but so far, the answer is No.

Reply:
Attached is the log file from my latest local run of ctest on Ubuntu.  I ran this command:
`QT_QPA_PLATFORM=vnc xvfb-run ctest`
[LastTest.log](https://github.com/musescore/MuseScore/files/3080993/LastTest.log)

As you can see, none of my `qDebug` statements execute. (a search for "Odd Left" produces zero results).  My current workaround includes a check for zero margins, so it's more limited than before.  That means I can now create an mtest for pagesettings.ui, so long I the test does not include setting the odd left margin specifically to zero.  That is a case that is easily avoided.
It's still a bummer that it must be worked around, but it's much less intrusive than my original workaround.

I'll run Travis in debug mode on my repo and get the log file from this latest push, so we have current data.  That will take a little while.  I'll post it here later.